documentation/docs/extensibility.md

```diff
@@ -1,86 +1,184 @@
 
-!!! warning "Decoupling"
-    If you go this route you will be decoupled from the innovations provided with project "Piper", unless you re-use for example stages (as indicated above under _2. Central Custom Templates_).
+This might be done for an individual project (in the `Jenkinsfile`), or in a separate git repository so it can be used for multiple projects.
 
-    **We recommend to use this only as last option for extensibility.**
+### Single project
 
+The default `Jenkinsfile` of centrally maintained pipelines does nothing except for loading the pipeline and running it.
+This is comfortable, but limits what you can modify.
+For example, you can't reorder stages, change which stages run in parallel or add new stages.
 
-## Further tips and information
+You can copy and paste the existing Pipelines from those sources into your `Jenkinsfile` and make changes to it
 
-When you consider to add additional capabilities your first stop should be the [Jenkins Pipeline Steps Reference](https://jenkins.io/doc/pipeline/steps/).
-Here you get an overview about what kind of capabilities are already available and a list of related parameters which you can use to customize the existing implementation. The provided information should help you to understand and extend the functionality of your pipeline.
+* [piperPipeline](https://github.com/SAP/jenkins-library/blob/master/vars/piperPipeline.groovy)
+* [SAP Cloud SDK Pipeline](https://github.com/SAP/cloud-s4-sdk-pipeline-lib/blob/master/vars/cloudSdkPipeline.groovy)
 
-!!! tip
-    If you consider extensions we recommend you to do it using a custom library according to the [Jenkins shared libraries](https://jenkins.io/doc/book/pipeline/shared-libraries/) concept instead of adding groovy coding to the `Jenkinsfile`.
-    Your custom library can easily live next to the provided pipeline library.
+### Multiple projects
 
-    Your Jenkinsfile would then start like
+Similar to what you can do in an individual `Jenkinsfile`, you can also copy the pipeline to a file in a separate git repository and modify it.
 
-    ```
-    @Library(['piper-lib-os', 'your-custom-lib']) _
+To do this, create a new git repository in your preferred git hosting service.
+It must be compliant to [how Jenkins shared libraries are built](https://jenkins.io/doc/book/pipeline/shared-libraries/).
+In a nutshell, this means you need a `vars` directory inside which you can place a copy of your preferred pipeline.
 
-    ```
+A minimal example of such a library could have this directory structure:
+
+```
+./vars/myCustomPipeline.groovy
+./README.md
+```
+
+where `myCustomPipeline.groovy` contains the modified pipeline code.
+
+!!! note
+    Your custom pipeline _must_ be named different from the other pipelines provided by project "Piper", because Jenkins requires names across multiple libraries to be unique.
```

Comment:
```suggestion
    Your custom pipeline _must_ be named differently from the other pipelines provided by project "Piper", because Jenkins requires names across multiple libraries to be unique.
```